src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/CodeStyleHelpers.cs

```diff
@@ -81,7 +81,7 @@ public static bool TryParseBoolEditorConfigCodeStyleOption(string arg, CodeStyle
                 {
                     var firstValue = arg[..firstColonIndex];
                     value = firstValue.Trim();
-                    notification = localNotification;
+                    notification = localNotification.WithIsExplicitlySpecified(true);
```

Comment:
CodeStyle option's `Notification` property now tracks if the severity was explicitly specified with `option_name = option_value:severity` or is the default severity as the entry was `option_name = option_value`